Objective-C/CBLReplicator.mm

```diff
@@ -433,37 +466,39 @@ - (void) c4StatusChanged: (C4ReplicatorStatus)c4Status {
             [self stopReachabilityObserver];
         }
         
-        // Prevent self to get released when removing from the active replications:
-        CBLReplicator* repl = self;
-        if (c4Status.level == kC4Stopped) {
-            CBL_LOCK(_config.database) {
-                // Remove from the active replications before posting change:
-                [_config.database.activeReplications removeObject: repl];
-            }
-        }
+        // Change from starting to started state:
+        if (_state == kCBLStateStarting && c4Status.level != kC4Stopped)
+            _state = kCBLStateStarted;
         
-        // Update my properties:
-        [self updateStateProperties: c4Status];
+        // Record raw status:
+        _rawStatus = c4Status;
         
-        // Post change
-        [_changeNotifier postChange: [[CBLReplicatorChange alloc] initWithReplicator: self
-                                                                              status: self.status]];
-        
-        // Clear replicator:
-        if (c4Status.level == kC4Offline && _isSuspending) {
-            _isSuspending = NO;
+        // Offline:
+        BOOL unsuspend = NO;
+        if (c4Status.level == kC4Offline) {
```

Comment:
this means, whenever the c4Stats is offline, we release the replicator?? Old code, we had `&& _isSuspending`.